src/Participants/Participant.java

```diff
@@ -1,7 +1,6 @@
 package Participants;
 
 public interface Participant {
-    void run(Racetrack obstacle);
-    void jump(Wall obstacle);
-    boolean overcome(Obstacle obstacle);
+    boolean run(Racetrack obstacle);
+    boolean jump(Wall obstacle);
```

Comment:
In this case for each new obstacle new method should be added, but they should do the same. Consider using just a distance/heigth for run/jump to not depend on obstacle implementation